File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
# def reverse_string(str)
#   # type your code in here
#   return str.reverse
# end

def reverse_string(str)
  reversed_str = ""

  str.chars.each do |char|
    reversed_str = char + reversed_str
    puts char, reversed_str
  end

  reversed_str
end

 puts reverse_string('pappy')
#  puts reverse_string('')
#  puts reverse_string('ta')

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')
  puts reverse_string('happy')
  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let myOut = str.split('').reverse()

  return myOut.join('')
}

console.log(reverseString('bye'))
console.log(reverseString('AwwwSnap'))
console.log(reverseString('racecar'))
console.log(reverseString(''))



if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution


